### PR TITLE
Extension process crash handling for persistent background pages

### PR DESCRIPTION
**Pull request by rebloor**
### Description

Address is the documentation requirements of [Bug 1848020](https://bugzilla.mozilla.org/show_bug.cgi?id=1848020) Changes to extension process crash handling for persistent background pages. Also addresses a number of minor text issues in the background script page.

### CODE REVIEW

File: files/en-us/mozilla/add-ons/webextensions/background_scripts/index.md [hunk 1]
```diff
@@ -13,6 +13,8 @@ Background scripts or a page are:
 - Persistent – loaded when the extension starts and unloaded when the extension is disabled or uninstalled.
 - Non-persistent (which are also known as event pages) – loaded only when needed to respond to an event and unloaded when they become idle. However, a background page does not unload until all visible views and message ports are closed. Opening a view does not cause the background page to load but does prevent it from closing.
 
+> **Note:** In Firefox, if the extension process crashes, an extension's persistent background scripts or pages that are enabled and running at the time of the crash are reloaded automatically. You can test this condition by opening a new tab and navigating to `about:crashextensions`, which triggers a crash of the extension process.
```
Comment (rpl) [inline files/en-us/mozilla/add-ons/webextensions/background_scripts/index.md:16]: only the persistent background page is reloaded automatically on extension process crashes (event pages are not restarted right away but they are being set back in the suspended state and so the event page will be respawned as expected when one of their persistent event listeners is going to be called).

All other extension pages are not reloaded automatically (e.g. an extension page loaded in a tab will not be reloaded automatically, a "tab crashed" UI would be rendered in the tab in that case).

As an example:
- with ublock origin installed (which is an extension that includes a persistent background page)
- open the browser action popup and click on the "Open the dashboard" gear button from ublock browser action popup => a new tab is opened to the ublock dashboard panel rendered in it
- after that open a new tab and navigate to about:crashextensions
- right after that switch back to the tab where the ublock dashboard panel was open and expect it to have a crash report UI rendered in it (with "Gah. Your tab just crashed" message and the "Close Tab" and "Restore This Tab" action buttons)

on the contrary ublock persistent background page has been reloaded automatically under the hood (if it wasn't ublock browser action panel would not work anymore).

**Review by rpl — CHANGES_REQUESTED** [review]

File: files/en-us/mozilla/add-ons/webextensions/background_scripts/index.md (hunk 1 above)
Comment (rebloor) [reply files/en-us/mozilla/add-ons/webextensions/background_scripts/index.md]: @rpl Curiously, when I followed your example in the general release version of Firefox, the page didn't crash even after I disabled all the other extensions I have installed. On the other hand, running in Firefox Developer Edition, it behaved as you described. I've pushed that update.

File: files/en-us/mozilla/add-ons/webextensions/background_scripts/index.md [hunk 2]
```diff
@@ -13,6 +13,10 @@ Background scripts or a page are:
 - Persistent – loaded when the extension starts and unloaded when the extension is disabled or uninstalled.
 - Non-persistent (which are also known as event pages) – loaded only when needed to respond to an event and unloaded when they become idle. However, a background page does not unload until all visible views and message ports are closed. Opening a view does not cause the background page to load but does prevent it from closing.
 
+> **Note:** In Firefox, if the extension process crashes, persistent background scripts running at the time of the crash are reloaded automatically. Open event pages aren't automatically restored. Event pages issue a warning to inform the user they have crashed and allow the user to close or restore the tab containing the page.
```
Comment (rpl) [inline files/en-us/mozilla/add-ons/webextensions/background_scripts/index.md:16]: The event page is an hidden page like the background page one, the only difference is that it is not persistent and instead suspended when idle and respawned when we are about to call a WebExtensions API event listener they have registered the previous one they have been executed, and so we'd need to tweak this some more, eg. something along these lines:

> **Note:** In Firefox, if the extension process crashes, persistent background scripts running at the time of the crash are reloaded automatically, non persistent background scripts (also known as "Event Pages") are not reloaded right away but they will be restarted automatically when Firefox will be calling one of their WebExtensions API events listeners. Extension pages loaded in a tab are not automatically restored, a warning page informs the user they have crashed and allow the user to close or restore the tab containing the crashed extension page.

**Review by rpl — CHANGES_REQUESTED** [review]
@rebloor my previous message was likely not clear enough about what I was referring to with "event page" (non persistent background scripts, how the crash handling related to those differs from the persistent background page, and which parts of my previous comment were specifically referring to the extension pages loaded in a tab instead (which are the ones that will not be restored automatically at all and a crashing page expected to be rendered instead), I'm so sorry about that :-(

Below a comment about tweaks to consider on the paragraph were we are describing these behaviors (to fix the part describing the behavior for the extension pages loaded in a tab as the behavior of the event pages).

File: files/en-us/mozilla/add-ons/webextensions/background_scripts/index.md (hunk 1 above)
Comment (rpl) [reply files/en-us/mozilla/add-ons/webextensions/background_scripts/index.md]: > @rpl Curiously, when I followed your example in the general release version of Firefox, the page didn't crash even after I disabled all the other extensions I have installed. On the other hand, running in Firefox Developer Edition, it behaved as you described. I've pushed that update.

I wonder if there is a flag that is preventing that to happen also in the Firefox release channel. I'll double-check that.

**Review by rpl — APPROVED** [review]
@rebloor lgtm, thanks!

**Pull request merged**